### PR TITLE
Adding BERT

### PR DESCRIPTION
**Pull request by pawel-polyai**
I added BERT (english version) with summation of embeddings for ouput tokens.

### CODE REVIEW

File: baselines/vector_based.py [hunk 1]
```diff
@@ -5,10 +5,12 @@
 import shutil
 import tempfile
 
+import bert
```
Comment (matthen) [inline baselines/vector_based.py]: we need a way to make this import work- can we maybe add bert as a git submodule? or is there a pip package?

File: baselines/vector_based.py [hunk 2]
```diff
@@ -59,6 +61,71 @@ def encode(self, texts):
...
+
+    Args:
+        uri: (string) the tensorflow hub URI for the model.
+    """
+    def __init__(self, uri):
+        """Create a new `TfHubEncoder` object."""
+        self._session = tf.Session(graph=tf.Graph())
+        with self._session.graph.as_default():
+            glog.info("Loading %s model from tensorflow hub", uri)
+            self.embed_fn = tensorflow_hub.Module(uri, trainable=False)
```
Comment (matthen) [inline baselines/vector_based.py]: these all should be pseudo-private

File: baselines/vector_based.py [hunk 3]
```diff
@@ -59,6 +61,71 @@ def encode(self, texts):
...
+
+            init_ops = (
+                tf.global_variables_initializer(), tf.tables_initializer())
+        glog.info("Initializing graph.")
+        self._session.run(init_ops)
+
+    def encode(self, texts):
+        """Encode the given texts."""
+        label_list = [0]
+        MAX_SEQ_LENGTH = 128
```
Comment (matthen) [inline baselines/vector_based.py]: just make this lower case

File: baselines/vector_based.py (hunk 1 above)
Comment (pawel-polyai) [reply baselines/vector_based.py]: there is a pip package - I added it to the requirements list.

File: baselines/vector_based.py [hunk 4]
```diff
@@ -59,6 +61,86 @@ def encode(self, texts):
         return self._session.run(self._embeddings, {self._fed_texts: texts})
 
 
+class BERTHubEncoder(Encoder):
```
Comment (matthen) [inline baselines/vector_based.py]: Please rename this to just `BERTEncoder`

**Comment by matthen** [comment]
LGTM after rename to `BERTEncoder`, though I am still surprised about the results

**Pull request merged**